src/core/mesh/qgsmeshlayerrenderer.cpp

```diff
@@ -300,6 +303,7 @@ bool QgsMeshLayerRenderer::render()
   renderMesh();
   renderVectorDataset();
 
+  mReadyToCompose = true;
```

Comment:
Are mesh layers ever slow to redraw? I'm wondering if a similar approach as is done for vector layers would be beneficial here, where we set a timer when the rendering starts and if it isn't complete in 3 seconds then we set mReadyToComposer to true early so that the progressive rendering updates are shown

Reply:
I think it could be interested only for the mesh frame or arrows of the vectors datasets (not for streamlines or traces that are actually rendered a bit like raster, that is, by producing an image). But as these renderings can be disabled or set to be coarser, I don't know if it  is good to add more complexity only for these renderings.
I will look how it is complex to add this for arrow and mesh frame...